src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -971,6 +1017,84 @@ Subgraph::SubgraphExecutor::SubgraphExecutor(const std::shared_ptr<Subgraph::Sub
 #endif
 }
 
+#ifdef OPENVINO_ARCH_X86_64
+std::vector<MemoryPtr> Subgraph::SubgraphExecutor::separately_repack_inputs(const dnnl::stream& strm,
+                                                                            const std::vector<MemoryPtr>& srcMemPtrs) {
+    auto get_batch_stride = [](const std::vector<size_t> strides) {
+        for (size_t i = 2; i < strides.size(); ++i)
+            if (*(strides.rbegin() + i) != 0)  // handle broadcasting pattern
+                return *(strides.rbegin() + i);
+        return (*++strides.rbegin());
+    };
+
+    auto reordered_in_ptrs = srcMemPtrs;
+    size_t offset = m_internal_buffer_size;
+    for (const auto& p : m_repacked_inputs) {
+        const auto in_idx = p.first;
+        const auto& repacked_input = p.second;
+        const auto& desc = repacked_input.desc;
+        const void* data_ptr = m_buffer_scratchpad->getDataAs<uint8_t>() + offset;
+
+        OPENVINO_ASSERT(in_idx < srcMemPtrs.size(), "Incorrect index of input repacked mem ptr");
+        const auto& src_mem = srcMemPtrs[in_idx];
+        const auto& dst_mem = std::make_shared<Memory>(strm.get_engine(), desc, data_ptr, false);
+
+        const auto& shape = dst_mem->getShape().getDims();
+        const auto batch = std::accumulate(shape.rbegin() + 2, shape.rend(), 1lu, std::multiplies<size_t>());
+        const auto in_stride = get_batch_stride(repacked_input.in_offsets);
+        const auto out_stride = get_batch_stride(repacked_input.out_offsets);
+
+        const auto* src = src_mem->getDataAs<const uint8_t>();
+        auto* dst = dst_mem->getDataAs<uint8_t>();
+
+        const auto& executor = repacked_input.executor;
+        parallel_for(batch, [&](size_t b0) {
+            BrgemmCopyBKernel::call_args args;
+            args.src = src + b0 * in_stride + m_start_offset_in[in_idx];
+            args.tr_src = dst + b0 * out_stride;
+            BrgemmCopyBKernelExecutor::execute(executor.get(), &args);
+        });
+
+        reordered_in_ptrs[in_idx] = dst_mem;
+        offset += desc->getCurrentMemSize();
+    }
+    return reordered_in_ptrs;
+}
+
+void Subgraph::SubgraphExecutor::in_parallel_repack_inputs(const std::vector<MemoryPtr>& inMemPtrs,
+                                                           const std::vector<size_t>& indexes,
+                                                           int ithr,
+                                                           jit_snippets_call_args& call_args) {
+    for (const auto& p : m_repacked_inputs) {
+        const auto& in_idx = p.first;
+        const auto& repacked_in = p.second;
+
+        const auto& src_offsets = repacked_in.in_offsets;
+        const auto& dst_offsets = repacked_in.out_offsets;
+
+        size_t src_offset = m_start_offset_in[in_idx], dst_offset = 0;
+        for (size_t j = 0; j < indexes.size(); j++) {
+            src_offset += src_offsets[j] * indexes[j];
+            dst_offset += dst_offsets[j] * indexes[j];
+        }
```

Comment:
This loop can't be vectorized because indexes.size() is not known at runtime. That's a pity, because we know that `indexes.size() == 6` in  96% cases. Of course it's hardly a bottleneck, but its quite easy to allow for this optimization - again templates 🙂 : indexes size must be a template parameter.
it also makes sense because input ranks can't change in runtime, so there will likely be only one instantiation of this template

Reply:
Create std::function which is inited in constuctor.
As far as I know, template parameter should be known in code compilation stage. So I decided to create `std::function` and just call in the needed code part.
Please take a look one more time!